resources/asciidoctor/lib/alternative_language_lookup/alternative.rb

```diff
@@ -10,43 +10,42 @@ class Alternative
       Alternative language must be a code block followed optionally by a callout list
     LOG
 
-    def initialize(listing, lang, dir, basename)
-      @listing = listing
+    def initialize(document, lang, path)
+      @document = document
       @lang = lang
-      @dir = dir
-      @basename = basename
-      @counter = listing.document.attr 'alternative_language_counter', 0
-      @loaded = false
+      @path = path
+      @counter = @document.attr 'alternative_language_counter', 0
+      @text = nil
       load
       return unless validate
 
       munge
-      @loaded = true
-      listing.document.attributes['alternative_language_counter'] = @counter + 1
+      @document.attributes['alternative_language_counter'] = @counter + 1
+      @text = @child.convert
```

Comment:
I don't know how expensive this operation is. If it is indeed expensive, maybe it's worth lazily calling `convert` in the `block` method, as it was before.

Reply:
The `load` and `convert` calls are comparatively slow, yeah. The way we use it this is fine though. I'll leave a note.